### PR TITLE
eliminate second app service

### PR DESCRIPTION
**Pull request by jplane**
was able to use a dedicated service principal (thx @charleszipp !) and go back to a single web app vs. two web apps

also was able to manually invoke the scale out/in endpoints

### CODE REVIEW

File: deploy.tf [hunk 1]
```diff
@@ -3,6 +3,10 @@ provider "azurerm" {
   features {}
 }
 
+locals {
+  sp_json = jsondecode(file("${path.module}/azureauth.json"))
```
Comment (ianphil) [inline deploy.tf:7]: You could pass this in using TF_VAR: https://www.terraform.io/docs/configuration/variables.html#environment-variables

File: frontend/ManagementController.cs [hunk 2]
```diff
@@ -91,19 +91,12 @@ private async Task CreateNewMetricsOutput(IAzure azure, string rg, string aci_na
...
-                                .WithResource(resourceId)
-                                .WithLogAnalytics(la_resource_Id)
-                                    .WithMetric("AllMetrics", TimeSpan.FromMinutes(1), 7)
-                                .CreateAsync();
-            }
-            catch(Exception ex)
-            {
-                _log.LogError(ex, "Azure error");
-            }
+            await azure.DiagnosticSettings
```
Comment (ianphil) [inline frontend/ManagementController.cs:94]: Why remove the `try/catch` ?

File: frontend/ManagementController.cs [hunk 3]
```diff
@@ -181,8 +174,14 @@ private AzureCredentials GetAzureCredentials()
             }
             else
             {
-                return factory.FromSystemAssignedManagedServiceIdentity(MSIResourceType.AppService,
-                                                                        AzureEnvironment.AzureGlobalCloud);
+                var tenantId = _config["SERVICE_PRINCIPAL_TENANT_ID"];
```
Comment (ianphil) [inline frontend/ManagementController.cs:177]: Should keep the MSI logic here behind a flag or something once the bug is resolved?

**Review by ianphil — APPROVED** [review]
Couple of questions, but looks good overall.

**Review by charleszipp — APPROVED** [review]

File: deploy.tf (hunk 1 above)
Comment (jplane) [reply deploy.tf:7]: true, tho create-service-principal.sh is used to write creds to a specific local file, so i think this is okay... ?

File: frontend/ManagementController.cs (hunk 2 above)
Comment (jplane) [reply frontend/ManagementController.cs:94]: it was there for debugging... TODO item exists for handling errors gracefully, in a more holistic manner

File: frontend/ManagementController.cs (hunk 3 above)
Comment (jplane) [reply frontend/ManagementController.cs:177]: i wouldn't expect the app service bug to be fixed anytime soon. this current solution is fine for prod, i think (once we get secrets into KV, which is another TODO item)

**Pull request merged**